src/main/kotlin/me/gserv/fabrikommander/data/TpaRequest.kt

```diff
@@ -0,0 +1,47 @@
+package me.gserv.fabrikommander.data
+
+import me.gserv.fabrikommander.utils.*
+import net.minecraft.server.network.ServerPlayerEntity
+import net.minecraft.text.ClickEvent
+import net.minecraft.text.HoverEvent
+import net.minecraft.text.MutableText
+import net.minecraft.util.Util
+
+class TpaRequest(
+    val source: ServerPlayerEntity,
+    val target: ServerPlayerEntity,
+    val availableForTicks: Int = 6000,
+    val tpaHere: Boolean
+) {
+    fun apply() {
+        if (tpaHere) target.teleport(source.serverWorld, source.x, source.y, source.z, source.yaw, source.pitch)
+        source.teleport(target.serverWorld, target.x, target.y, target.z, target.yaw, target.pitch)
+    }
+
+    fun notifyTarget() {
+        // Message will be configurable later
+        val message = click(
```

Comment:
I would prefer this text to be more like `User has requested to teleport to you. [Accept / Deny]`